src/bitcoin-cli.cpp

```diff
@@ -191,7 +192,9 @@ static void http_error_cb(enum evhttp_request_error err, void *ctx)
 
 UniValue CallRPC(const std::string& strMethod, const UniValue& params)
 {
-    std::string host = GetArg("-rpcconnect", DEFAULT_RPCCONNECT);
+    std::string host;
+    int dummy_port;
+    SplitHostPort(GetArg("-rpcconnect", DEFAULT_RPCCONNECT), dummy_port, host);
```

Comment:
I chose not to allow overriding the port since we already have the `-rpcport` option, and this is the minimal functional change.

I'm equally happy to make the one-line change so `-rpcconnect` can accept `<address>:<port>`. I have zero preference either way and I'm happy to go along with consensus opinion. @theuni has already expressed slight preference for allowing a port number in `-rpcconnect`.